llama_index/query_engine/sql_join_query_engine.py

```diff
@@ -161,7 +163,7 @@ def __init__(
         self,
         sql_query_tool: QueryEngineTool,
         other_query_tool: QueryEngineTool,
-        selector: Optional[LLMSingleSelector] = None,
+        selector: Optional[Union[LLMSingleSelector, PydanticSingleSelector]] = None,
```

Comment:
very nit: I think we could type as `BaseSelector`?

Reply:
I think it specifically has to be a single selector though?